tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -1879,4 +1879,36 @@ protected function assertProxyMethod($validator, $method, $extra = null, $pass =
         $this->assertEquals('the message', $rule->get('message'), 'Error messages are not the same');
         $this->assertEquals('create', $rule->get('on'), 'On clause is wrong');
     }
+
+    /**
+     * Testing adding DefaultProvider
+     *
+     * @return void
+     */
+    public function testAddingDefaultProvider()
+    {
+        $validator = new Validator;
+        $this->assertEmpty($validator->providers(), 'Providers should be empty');
+
+        Validator::addDefaultProvider('test-provider', '\MyNameSpace\Validation\MyProvider');
+        $validator = new Validator;
+        $this->assertEquals($validator->providers(), ['test-provider'], 'Default provider `test-provider` is missing');
+    }
+
+    /**
+     * Testing getting DefaultProvider(s)
+     *
+     * @return void
+     */
+    public function testGetDefaultProvider()
+    {
+        Validator::addDefaultProvider('test-provider', '\MyNameSpace\Validation\MyProvider');
+        $this->assertEquals(Validator::getDefaultProvider('test-provider'), '\MyNameSpace\Validation\MyProvider', 'Default provider `test-provider` is missing');
+
+        $this->assertNull(Validator::getDefaultProvider('invalid-provider'), 'Default provider (`invalid-provider`) should be missing');
+        $this->assertNull(Validator::getDefaultProvider(null), 'Default provider (null) should be missing');
+        
```

Comment:
I normally format the code before hand but the default formatting is so different from your codesniffer rules. Like it wraps long lines but not in the same way as PHPCS wants ;-)